src/opengl/nodes/text.rs

```diff
@@ -0,0 +1,133 @@
+//! The text node draws text at a specified position and in a specified color
+
+use failure::Error;
+use glium::backend::Facade;
+use glium::texture::{RawImage2d, Texture2d};
+use glium::Surface;
+use image;
+use owning_ref::OwningHandle;
+use std::path::Path;
+use std::rc::Rc;
+
+use opengl::{MapAsUniform, UniformsStorageVec};
+use opengl::text::TextRenderer;
+use super::Node;
+use util::DerefInner;
+
+/// A node that draws text
+pub struct TextNode {
+    /// The name of the node
+    name: String,
+    /// The Facade it uses to work with the OpenGL context
+    facade: Rc<Facade>,
+    /// The inner texture it renders to
+    texture: Rc<Texture2d>,
+    /// The TextRenderer it uses to render text
+    text_renderer: TextRenderer,
+    /// The text it draws
+    text: String,
+    /// The position to draw the text
+    position: [f32; 2],
+    /// The color of the text in RGBA format
+    color: [f32; 4],
+}
+
+impl TextNode {
+    /// Create a new instance
+    pub fn new(
+        facade: &Rc<Facade>,
+        name: String,
+        text: String,
+        position: [f32; 2],
+        color: [f32; 4],
+        font_name: &str,
+        font_size: f32,
+    ) -> Result<Self, Error> {
+        let (width, height) = facade.get_context().get_framebuffer_dimensions();
+        let texture = Rc::new(Texture2d::empty(&**facade, width, height)?);
+
+        let text_renderer = TextRenderer::new(facade.clone(), font_name, font_size)?;
+
+        Ok(Self {
+            name,
+            facade: Rc::clone(facade),
+            texture,
+            text_renderer,
+            text,
+            position,
+            color,
+        })
+    }
+
+    /// Set the text
+    pub fn set_text(&mut self, text: String) {
```

Comment:
this is very fine but in the future perhaps we should consider setter macros!?

Reply:
Agreed!